src/lib.rs

```diff
@@ -60,6 +60,16 @@ pub use fields::TemporalFields;
 /// The `Temporal` result type
 pub type TemporalResult<T> = Result<T, TemporalError>;
 
+#[macro_export]
+macro_rules! temporal_assertion {
+    ($assertion:expr) => {{
+        let Some(value) = $assertion else {
+            return Err(TemporalError::assert());
+        };
+        value
+    }};
+}
+
```

Comment:
This makes deconstruction a bit more opaque, which I don't prefer.

What about a `TemporalUnwrap` trait implemented for `Option<T>? It could be interesting that way, since we could extend it in the future with a `context` parameter if we need to.

Reply:
Oooh, that's a nice idea. Give me a few minutes to put it together.